demo/android/OrcaDemo/orca-demo-app/src/main/java/ai/picovoice/orcademo/MainActivity.java

```diff
@@ -322,25 +349,190 @@ private void stopPlayback() {
             }
         } else {
-            stopPlayback();
+            runStreamSynthesis(text);
         }
     }
 
+    private void stopStreamPlay() {
+        try {
+            runOnUiThread(() -> {
+                setUIState(UIState.EDIT);
+                infoTextView.setText("");
+                streamTextView.setVisibility(View.INVISIBLE);
+                streamSecsTextView.setVisibility(View.INVISIBLE);
+                synthesizeEditText.setVisibility(View.VISIBLE);
+            });
+        } catch (Exception e) {
+            displayError(e.toString());
+        }
+    }
+
+    private void runStreamSynthesis(final String text) {
+        runOnUiThread(() -> {
+            setUIState(UIState.PLAYBACK);
+            infoTextView.setText("Streaming...");
+            streamTextView.setVisibility(View.VISIBLE);
+            streamSecsTextView.setVisibility(View.VISIBLE);
+            synthesizeEditText.setVisibility(View.INVISIBLE);
+        });
+
+        AtomicBoolean isStreamingText = new AtomicBoolean(false);
+        ArrayList<String> textStream = new ArrayList<>();
+
+        executor.submit(() -> {
+            isStreamingText.set(true);
+
+            String[] words = text.split(" ");
+            for (String word : words) {
+                word += " ";
+                try {
+                    textStream.add(word);
+                    Thread.sleep(100);
+                    streamTextView.append(word);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            isStreamingText.set(false);
+        });
+
+        AtomicBoolean isQueueingStreamingPcm = new AtomicBoolean(false);
+        ConcurrentLinkedQueue<short[]> pcmQueue = new ConcurrentLinkedQueue<>();
+        CountDownLatch latch = new CountDownLatch(1);
+
+        executor1.submit(() -> {
+            try {
+                mainHandler.post(() -> {
+                    streamTextView.setText("");
+                    streamSecsTextView.setText("");
+                    synthesizeButton.setEnabled(false);
+                });
+
+                int numIterations = 0;
+                boolean isPcmPlayStarted = false;
+                float secs = 0;
+                isQueueingStreamingPcm.set(true);
+
+                while (isStreamingText.get() || !textStream.isEmpty()) {
+                    if (!textStream.isEmpty()) {
+                        String word = textStream.remove(0);
+                        try {
+                            short[] pcm = orcaStream.synthesize(word);
+                            if (pcm != null && pcm.length > 0) {
+                                pcmQueue.add(pcm);
+                                secs += (float) pcm.length / orca.getSampleRate();
+                                streamSecsTextView.setText("Seconds of audio synthesized: " + String.format("%.3f", secs) + "s");
+                                if (numIterations == 1) {
```

Comment:
Let's define that somewhere more visible and call it something like: `num_audio_wait_chunks = 1`